pal/input/native/mouse-input.ts

```diff
@@ -95,9 +99,14 @@ export class MouseInputSource {
             const eventMouse = new EventMouse(eventType, false, this._preMousePos, mouseEvent.windowId);
             eventMouse.setLocation(location.x, location.y);
             eventMouse.setButton(button);
-            eventMouse.movementX = location.x - this._preMousePos.x;
-            eventMouse.movementY = this._preMousePos.y - location.y;
-
+            if (this._pointLocked && eventType === InputEventType.MOUSE_MOVE) {
+                const dpr = screenAdapter.devicePixelRatio;
+                eventMouse.movementX = mouseEvent.xref === undefined ? 0 : mouseEvent.xref * dpr;
+                eventMouse.movementY = mouseEvent.yref === undefined ? 0 : mouseEvent.yref * dpr;
```

Comment:
it's safer to use `typeof mouseEvent.xref === 'undefined'` instead of `mouseEvent.xref === undefined`